packages/babel-helper-define-polyfill-provider/src/imports-cache.ts

```diff
@@ -1,8 +1,5 @@
-// @flow
-
 import type { NodePath } from "@babel/traverse";
-import * as babel from "@babel/core";
-const { types: t } = babel.default || babel;
+import { types as t } from "@babel/core";
```

Comment:
so the previous version was erroring in TS.

per my commit message, `esModuleInterop` in `tsconfig.json` should normally handle this automatically... but we're building with `@babel/preset-typescript`, not TS, which, if I recall correctly, does not use `tsconfig.json`, so I'm not sure if this holds entirely

Reply:
The problem is that we are compiling this file both to CJS and to ESM: when it runs directly in Node.js, the import is only accessible as `babel.default.types`. The best thing is probably to do something like this to make every tool happy:
```js
const { types: t } = (babel.default || babel) as typeof import("@babel/core");
```
and stick a `@ts-expect-error` in front of it if TS complains.

Reply:
I see now that the unit tests pass that `test:esm` fails due to this. 

When compiling with Babel, is there no equivalent of TS's `esModuleInterop` (and the whole `__esModule` property etc) that would make it compatible with both without such a hack?

Reply:
This would require a `// @ts-expect-error` everywhere as it does still error due to "Property 'default' does not exist on type... "

The cast does fix `t`'s type so that it doesn't interpret as `any` due to that error.

Reply:
Nicolò wrote up a tiny Babel plugin to workaround this in https://github.com/babel/babel-polyfills/pull/121/commits/68f27965a2d2d3779bda0f8d1e33b68346c291b4